static/js/publisher/release/selectors/index.js

```diff
@@ -261,3 +261,34 @@ export function getRevisionsFromBuild(state, buildId) {
     revision => getBuildId(revision) === buildId
   );
 }
+
+// return the progressive release status based on channel, arch and optional revision
+export function getProgressiveState(state, channel, arch, revision) {
+  const { releases } = state;
+  const release = releases.filter(item => {
+    if (
+      channel === getChannelString(item) &&
+      arch === item.architecture &&
+      item.revision &&
+      (!revision || item.revision === revision)
+    ) {
+      return item;
+    }
+    return false;
+  });
+
+  let progressive = null;
+
+  if (
+    release.length > 1 &&
+    release[0] &&
+    release[0] &&
+    release[0].progressive &&
+    release[0].progressive.key
+  ) {
+    progressive = release[0].progressive;
+    progressive.from = release[1].revision;
```

Comment:
I'm not sure if I understand here.

The `from` prop is meant to be revision we are "progressing" from. But, if `revision` is passed as param to this function, then `release` array will contain only items where `item.revision` equals `revision` from the param, so... `release[1].revision` will also (?).

I don't understand why we have the optional `revision` param to this function, what does it change? When we want to use it, and when not?
Tests seem to cover only calling function without revision, but in the code we seem to pass current revision to it...

Reply:
Ah yes, this could definitely fail in some situations - I'll update the tests.
I added the optional `revision` for the history table. Simply asking if the `channel` and `arch` matched returned true for every revision in the history, the revision is meant to be a "is this specific revision being phased". Perhaps it doesn't need to be optional. I'll take a look.

Reply:
@Lukewh
It's fine if this function is both "give me progressive state for given channel/arch" (if revision is not included) or "give me progressive state of given revision in given channel/arch" (if revision is given).

But the way the `.from` is calculated is weird (or maybe just wrong) when there will be 2 progressive releases one after another of the same revision (when % is increased for example):

```
{
          architecture: "arch2",
          branch: null,
          track: "latest",
          risk: "stable",
          revision: "3",
          progressive: {
            key: "test",
            percentage: 60,
            paused: false
          }
        },
        {
          architecture: "arch2",
          branch: null,
          track: "latest",
          risk: "stable",
          revision: "3",
          progressive: {
            key: "test",
            percentage: 50,
            paused: false
          }
        },
        {
          architecture: "arch2",
          branch: null,
          track: "latest",
          risk: "stable",
          revision: "2",
          progressive: null
        }
```

I don't think we can assume that `release[1]` is the 'previous' revision. It can be previous release of the same revision. Especially if `revision` is passed as a parameter and releases are filtered based on it (all filtered items will be about only this revision).